quipucords/fingerprinter/jboss_brms.py

```diff
@@ -76,6 +76,14 @@ def classify_version_string(version_string):
     return None
 
 
+def _get_val_or_empty_list(data_dict, key):
```

Comment:
Is this going to be called whenever we don't scan middleware ?

Reply:
`detect_jboss_brms` will be called whether we scan middleware or not.
then it would use default dict getter for many keys with default set to `list()` or `set()`

This will be used later based on the premise that even if nothing is found an iterable will be there.
Changing the facts to always be filled with `None` breaks this behavior because the value `None` will take precedence over the default provided to the getter.

In other words...
```python
data_dict1 = {}
manifest_mfs1 = data_dict1.get(JBOSS_BRMS_MANIFEST_MF, set())
assert type(manifest_mfs1) == type(set())

data_dict2 = {JBOSS_BRMS_MANIFEST_MF: None}
manifest_mfs2 = data_dict2.get(JBOSS_BRMS_MANIFEST_MF, set())
assert type(manifest_mfs2) == type(None)
```